src/docfx/resolve/Resolve.cs

```diff
@@ -2,82 +2,88 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
 using System.IO;
 
 namespace Microsoft.Docs.Build
 {
-    internal delegate (string content, Document path) ResolveContent(Document relativeTo, string href);
-
-    internal delegate string ResolveHref(Document relativeTo, string href, Document resultRelativeTo);
-
     internal static class Resolve
     {
         public static (string content, Document file) TryResolveContent(this Document relativeTo, string href)
         {
-            var buildItem = TryResolveHref(relativeTo, href);
-            if (buildItem == null)
-            {
-                return default;
-            }
+            var (file, _) = TryResolveFile(relativeTo, href);
 
-            return (buildItem.ReadText(), buildItem);
+            return file != null ? (file.ReadText(), file) : default;
         }
 
         public static (string href, Document file) TryResolveHref(this Document relativeTo, string href, Document resultRelativeTo = null)
         {
-            var file = TryResolveHref(relativeTo, href);
+            var (file, fragmentQuery) = TryResolveFile(relativeTo, href);
+
+            // Cannot resolve the file, leave href as is
             if (file == null)
             {
                 return (href, null);
             }
 
-            var (_, fragment, query) = HrefUtility.SplitHref(href);
-
-            var resolvedHref = Uri.EscapeUriString(file.SiteUrl) + fragment + query;
-            if (href[0] == '/')
-            {
-                return (resolvedHref, file);
-            }
+            var resolvedHref = file.SiteUrl + fragmentQuery;
 
             if (resultRelativeTo == null)
             {
                 return (resolvedHref, file);
             }
 
+            // Make result relative to `resultRelativeTo`
             resolvedHref = PathUtility.GetRelativePathToFile(resultRelativeTo.SiteUrl, file.SiteUrl).Replace('\\', '/');
 
-            return (Uri.EscapeUriString(resolvedHref) + fragment + query, file);
+            return (resolvedHref + fragmentQuery, file);
         }
 
-        private static Document TryResolveHref(this Document relativeTo, string href)
+        private static (Document file, string fragmentQuery) TryResolveFile(this Document relativeTo, string href)
```

Comment:
>TryResolveFile [](start = 61, length = 14)

TryResolveHref? as the parameter is href

Reply:
that'll have conflict with the above TryResolveHref. The naming suggest the return value, so TryResolveHref returns an href, TryResolveFile returns a document, TryResolveContent returns the text.

---
In reply to: [193278780](https://github.com/dotnet/docfx/pull/2839#discussion_r193278780) [](ancestors = 193278780)